### PR TITLE
Ported the ACE2 ear ringing logic

### PR DESCRIPTION
**Pull request by ulteq**

### CODE REVIEW

**Comment by jonpas** [comment]
This fixed #619 ?

**Comment by esteldunedain** [comment]
> This fixes #619

Yes it does

File: addons/hearing/functions/fnc_firedNear.sqf [hunk 1]
```diff
@@ -47,10 +47,18 @@ if (_silencer != "") then {
     _audibleFireCoef = getNumber (configFile >> "CfgWeapons" >> _silencer >> "ItemInfo" >> "AmmoCoef" >> "audibleFire");
 };
 
-_audibleFire = getNumber (configFile >> "CfgAmmo" >> _ammo >> "audibleFire");
-
-_loudness = _audibleFireCoef * _audibleFire / 64;
-_strength = _vehAttenuation * (_loudness - (_loudness/50 * _distance)); // linear drop off
+_magazine = (getArray(configFile >> "CfgWeapons" >> _weapon >> "magazines")) select 0;
```
Comment (esteldunedain) [inline addons/hearing/functions/fnc_firedNear.sqf]: This line is a bit fishy. The first magazine supported by the weapon may not be the one that fired, which may be even more true when magazineGroups are used.

File: addons/hearing/functions/fnc_firedNear.sqf [hunk 2]
```diff
@@ -47,10 +47,18 @@ if (_silencer != "") then {
...
+_initSpeed = getNumber(configFile >> "CfgMagazines" >> _magazine >> "initSpeed");
+_ammoConfig = (configFile >> "CfgAmmo" >> _ammo);
+_caliber = getNumber(_ammoConfig >> "ACE_caliber");
+_parentClasses = [_ammoConfig, true] call BIS_fnc_returnParents;
+if ("RocketBase" in _parentClasses || "MissileBase" in _parentClasses) then {
+    _caliber = 200;
+} else {
+    if (_caliber <= 0) then { _caliber = 6.5; };
+};
+_loudness = (_caliber ^ 1.25 / 10) * (_initspeed / 1000) * _audibleFireCoef / 5;
```
Comment (esteldunedain) [inline addons/hearing/functions/fnc_firedNear.sqf]: loudness also needs to take into account the presence of supressors/silencers, which are attachments in A3.

**Pull request merged**